DigitalOffice/Services/UserService/Commands/UserCreateCommand.cs

```diff
@@ -0,0 +1,51 @@
+﻿using FluentValidation;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.ObjectModel;
+using System.Security.Cryptography;
+using System.Text;
+using UserService.Database;
+using UserService.Database.Entities;
+using UserService.Interfaces;
+using UserService.Repositories;
+using UserService.Repositories.Interfaces;
+using UserService.RestRequests;
+
+namespace UserService.Commands
+{
+    public class UserCreateCommand : IUserCreateCommand
+    {
+        private readonly IValidator<UserCreateRequest> validator;
+        private readonly IUserRepository repository;
+
+        public UserCreateCommand([FromServices] IValidator<UserCreateRequest> validator, [FromServices] IUserRepository repository)
+        {
+            this.validator = validator;
+            this.repository = repository;
+        }
+
+        public bool Execute(UserCreateRequest request)
+        {
+            validator.ValidateAndThrow(request);
+
+            var user = new User
+            {
+                Id = Guid.NewGuid(),
+                Email = request.Email,
+                FirstName = request.FirstName,
+                LastName = request.LastName,
+                MiddleName = request.MiddleName,
+                Status = request.Status,
+                PasswordHash = (new SHA512Managed().ComputeHash(Encoding.Default.GetBytes(request.Password))).ToString(),
+                AvatarFileId = null,
+                IsActive = true,
+                CertificatesFilesIds = new Collection<UserCertificateFile>(),
+                AchievementsIds = new Collection<UserAchievement>()
```

Comment:
Shouldn't there be a null check for unnecessary fields thatmight be missing in request?

Reply:
Nope, this is not needed